streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -219,26 +219,20 @@ public <K, V> void send(final String topic,
                 }
 
                 if (!topic.endsWith("-changelog")) {
-                    final Map<String, Sensor> producedSensorByTopic = sinkNodeToProducedSensorByTopic.get(processorNodeId);
-                    if (producedSensorByTopic == null) {
-                        log.error("Unable to records bytes produced to topic {} by sink node {} as the node is not recognized.\n"
-                                      + "Known sink nodes are {}.", topic, processorNodeId, sinkNodeToProducedSensorByTopic.keySet());
-                    } else {
-                        // we may not have created a sensor during initialization if the node uses dynamic topic routing,
-                        // as all topics are not known up front, so create the sensor for that topic if absent
-                        final Sensor topicProducedSensor = producedSensorByTopic.computeIfAbsent(
+                    // we may not have created a sensor during initialization if the node uses dynamic topic routing,
+                    // as all topics are not known up front, so create the sensor for this topic if absent
+                    final Sensor topicProducedSensor = producedSensorByTopic.computeIfAbsent(
+                        topic,
+                        t -> TopicMetrics.producedSensor(
+                            Thread.currentThread().getName(),
+                            taskId.toString(),
+                            processorNodeId,
                             topic,
-                            t -> TopicMetrics.producedSensor(
-                                Thread.currentThread().getName(),
-                                taskId.toString(),
-                                processorNodeId,
-                                topic,
-                                context.metrics()
-                            )
-                        );
-                        final long bytesProduced = producerRecordSizeInBytes(serializedRecord);
-                        topicProducedSensor.record(bytesProduced, context.currentSystemTimeMs());
-                    }
+                            context.metrics()
+                        )
+                    );
+                    final long bytesProduced = producerRecordSizeInBytes(serializedRecord);
+                    topicProducedSensor.record(bytesProduced, context.currentSystemTimeMs());
```

Comment:
A few unit tests fail because this change removes the if-else statement and the passed context is null. What I find more interesting, however, is that the tests do not fail on line 231 following the same reasoning 🤔

Reply:
To answer my own question: The `NullPointerException` surfaces on line 235 and not on line 231 because the topic already makes it into the Map as it is the sink and sinks are registered here: https://github.com/apache/kafka/pull/12836/files#diff-36dd8c03fa5252dbd39042bb49a0d6272728fbd46459c6a45f5189fa59749b32R97

Reply:
Ok I went through the code to double check whether it's possible for the context to actually be `null` in the non-testing code, and I'm pretty sure it should always have been set before we try to send any records -- probably we were just being lazy in the tests (although some of them do pass in a non-null context, so I'm not sure why others don't)

Anyways, I fixed the tests -- lmk if there's anything else, otherwise I'll merge this once the PR build finishes running